File: 02_algorithm/programmers/Level1/3진법 뒤집기.py
notation = '012'

def make_tri_number(num):
    mok, nmg = divmod(num, 3)
    n = notation[nmg]
    return make_tri_number(mok) + n if mok else n

def solution(n):
    tri_num = make_tri_number(n)
    reversed_tri_num = tri_num[::-1]
    return int(reversed_tri_num, 3)

print(solution(45))

File: 06_js/04_algorithm/programmers/Level1/3진법 뒤집기.js
function solution(n) {
  const notation = '012';
  const makeTriNumber = num => {
    const mok = parseInt(num / 3);
    const nmg = num % 3;
    const n = notation[nmg];
    return mok ? makeTriNumber(mok) + n : n;
  }
  
  const reversedTriNumber = makeTriNumber(n).split('').reverse().join('');
  return parseInt(reversedTriNumber, 3);
}

console.log(solution(45)); // 7
console.log(solution(125)); // 229